cedar-policy-core/src/evaluator/err.rs

```diff
@@ -83,24 +84,25 @@ pub enum EvaluationError {
     InvalidRestrictedExpression(#[from] RestrictedExpressionError),
 
     /// Thrown when a policy is evaluated with an un-filled slot
-    #[error("Template slot {0} was not instantiated")]
+    #[error("template slot `{0}` was not filled")]
     TemplateInstantiationError(SlotId),
 
     /// Evaluation error thrown by an extension function
-    #[error("error from {extension_name} extension: {msg}")]
+    #[error("error while evaluating {extension_name} extension function: {msg}")]
     ExtensionError {
         /// Name of the extension throwing the error
         extension_name: Name,
         /// Error message from the extension
         msg: String,
     },
 
-    /// Error raised if an expression did not reduce to a value when it was supposed to
-    #[error("The expression evaluated to a residual: {0}")]
+    /// This error is raised if an expression contains unknowns, but we weren’t
+    /// using the partial evaluation APIs.
```

Comment:
The "but we weren't..." part of the sentence is confusing to me. Do you mean that the `partial-eval` flag wasn't enabled?